src/metabase/query_processor/store.clj

```diff
@@ -208,14 +209,10 @@
     (let [provider   (metadata-provider)
           objects    (vec (if (satisfies? lib.metadata.protocols/BulkMetadataProvider provider)
                             (filter some? (lib.metadata.protocols/bulk-metadata provider metadata-type ids-set))
-                            (let [f (case metadata-type
-                                      :metadata/card    lib.metadata.protocols/card
-                                      :metadata/column  lib.metadata.protocols/field
-                                      :metadata/metric  lib.metadata.protocols/metric
-                                      :metadata/segment lib.metadata.protocols/segment
-                                      :metadata/table   lib.metadata.protocols/table)]
-                              (for [id ids-set]
-                                (f provider id)))))
+                            (lib.metadata.composed-provider/fetch-bulk-metadata-with-non-bulk-provider
+                             provider
+                             metadata-type
+                             ids-set)))
```

Comment:
Moved to somewhere more general. The QP Store is basically deprecated, and will be removed when the MLv2 QP project is complete. Don't need a QP Store when the query itself has a metadata provider attached